### PR TITLE
fix sitemap routing and update xml configuration schema

### PR DESCRIPTION
**Pull request by dbu**
started working on the doc, and found some issues.

we seem to lack the configuration test with xml schema (or need to add the defaults config into it)

### CODE REVIEW

File: DependencyInjection/CmfSeoExtension.php [hunk 1]
```diff
@@ -267,15 +267,18 @@ private function loadSitemapHandling($config, XmlFileLoader $loader, ContainerBu
...
             }
+            unset($configurations[$key]['default_change_frequency']);
+
             if (isset($config['defaults']['templates'])) {
                 // copy default configuration into this sitemap configuration to keep controller simple
                 foreach ($config['defaults']['templates'] as $format => $name) {
-                    if (!isset($configurations['templates'][$format])) {
-                        $configurations['templates'][$format] = $name;
+                    if (!isset($configurations[$key]['templates'][$format])) {
+                        $configurations[$key]['templates'][$format] = $name;
```
Comment (dbu) [inline DependencyInjection/CmfSeoExtension.php]: the configuration test seems to be incomplete, as this mistake was not detected.

File: Resources/config/routing/sitemap.xml [hunk 2]
```diff
@@ -4,9 +4,8 @@
     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
     xsi:schemaLocation="http://symfony.com/schema/routing http://symfony.com/schema/routing/routing-1.0.xsd">
 
-    <route id="cmf_seo_sitemap" pattern="/sitemap.{_format}">
+    <route id="cmf_seo_sitemap" pattern="/{sitemap}.{_format}">
         <default key="_controller">cmf_seo.sitemap.controller:indexAction</default>
-        <default key="sitemap">default</default>
```
Comment (dbu) [inline Resources/config/routing/sitemap.xml]: we only supported the configuration `default` in the large pull request.

File: Resources/config/schema/seo-1.0.xsd [hunk 3]
```diff
@@ -74,6 +75,14 @@
         <xsd:attribute name="default-change-frequency" type="change_frequency" default="always" />
     </xsd:complexType>
 
+    <xsd:complexType name="named_sitemap_configuration">
+        <xsd:complexContent>
+            <xsd:extension base="sitemap_configuration">
+                <xsd:attribute name="name" type="xsd:string" />
```
Comment (dbu) [inline Resources/config/schema/seo-1.0.xsd]: @WouterJ is this the correct way of doing it?

Comment (WouterJ) [inline Resources/config/schema/seo-1.0.xsd]: yes, afaik.

File: Resources/config/sitemap.xml [hunk 4]
```diff
@@ -29,31 +29,31 @@
 
         <service id="cmf_seo.sitemap.guesser_chain" class="%cmf_seo.sitemap.guesser_chain.class%" />
 
-        <service id="cmf_seo.sitemap.guesser.seo_metadata_title" class="%cmf_seo.sitemap.guesser.seo_metadata_title.class%" >
+        <service id="cmf_seo.sitemap.guesser.seo_metadata_title" class="%cmf_seo.sitemap.guesser.seo_metadata_title.class%" public="false">
             <argument type="service" id="cmf_seo.presentation" />
-            <tag name="cmf_seo.sitemap.guesser" />
+            <tag name="cmf_seo.sitemap.guesser" priority="-2"/>
```
Comment (dbu) [inline Resources/config/sitemap.xml]: using a priority that lets user provided guessers come first when they specify no explicit priority.

File: Sitemap/LocationGuesser.php [hunk 5]
```diff
@@ -41,14 +36,6 @@ public function __construct(UrlGeneratorInterface $router)
     }
 
     /**
-     * @param AlternateLocaleProviderInterface $alternateLocaleProvider
-     */
-    public function setAlternateLocaleProvider(AlternateLocaleProviderInterface $alternateLocaleProvider)
-    {
-        $this->alternateLocaleProvider = $alternateLocaleProvider;
-    }
```
Comment (dbu) [inline Sitemap/LocationGuesser.php]: removing dead code left over from splitting the guesser

**Comment by dbu** [comment]
@ElectricMaxxx i added some more things here. can you please review this one? i wrote the documentation assuming the changes in here - if we change things, help me remember to also adjust the doc.

**Comment by ElectricMaxxx** [comment]
Will have a look tonight.

File: Resources/config/sitemap.xml (hunk 4 above)
Comment (ElectricMaxxx) [inline Resources/config/sitemap.xml]: Isn't -2 smaler than 0? And wouldn't that let the users guesser come after the provided ones?

Comment (dbu) [inline Resources/config/sitemap.xml]: the higher the priority, the earlier a service comes in the chain.

i think i inverted that in one of my pull requests, it was lowest first
before, but both cmf routing and symfony core do highest first.

File: Resources/config/sitemap.xml [hunk 6]
```diff
@@ -15,7 +15,7 @@
         <parameter key="cmf_seo.sitemap.guesser.location.class">Symfony\Cmf\Bundle\SeoBundle\Sitemap\LocationGuesser</parameter>
         <parameter key="cmf_seo.sitemap.guesser.default_change_frequency.class">Symfony\Cmf\Bundle\SeoBundle\Sitemap\DefaultChangeFrequencyGuesser</parameter>
         <parameter key="cmf_seo.sitemap.voter_chain.class">Symfony\Cmf\Bundle\SeoBundle\Sitemap\VoterChain</parameter>
-        <parameter key="cmf_seo.sitemap.phpcr.publish_workflow_voter.class">Symfony\Cmf\Bundle\SeoBundle\Sitemap\PublishWorkflowVoter</parameter>
```
Comment (ElectricMaxxx) [inline Resources/config/sitemap.xml]: Did we move that voter out of the PHPCR scope? the publish workflow stuff is PHPCR, isn't it?

**Comment by ElectricMaxxx** [comment]
travis fails on `allowed failure` only?

File: Resources/config/sitemap.xml (hunk 6 above)
Comment (dbu) [inline Resources/config/sitemap.xml]: we moved it out. publish workflow is completely storage independant.

**Comment by dbu** [comment]
fixed the configuration test, now its hopefully green

**Comment by ElectricMaxxx** [comment]
looks good, thanks @dbu

**Pull request merged**